### ISSUE TITLE
Absolute number of failures instead of rate?

### ISSUE THREAD

**Issue by flozano** [javaslang/javaslang-circuitbreaker#6]
failureRateThreshold is a percentage. I'd like the circuit-breaker to be triggered immediately after the first failure... Could it be possible to have an absolute value for the failures, rater than a rate?

**Comment by flozano** [comment]
(version 0.2 had the very useful `maxFailures`)

**Comment by RobWin** [comment]
Hi, the new Version does not support it anymore, because it the previous solution not good when the backend system is just unreliable for a short time and not down. You can either use the old version or you could set `ringBufferSizeInHalfOpenState(1)` and `ringBufferSizeInClosedState(1)` .
The previous `maxFailures=3` could be `ringBufferSizeInHalfOpenState(3)` and `ringBufferSizeInClosedState(3)` and failureRateThreshold(100) .

**Comment by flozano** [comment]
OK, thanks for the explanation!

**Comment by RobWin** [comment]
You are welcome

**Comment by flozano** [comment]
I'm sorry for reopening, but I think there is something wrong...

In order to get same behaviour as in previous `maxFailures(N)`, I have to do `.ringBufferSizeInHalfOpenState(1).ringBufferSizeInClosedState(N)`, and not include `failureRateThreshold(100)`. 

This passes:
```
	@Test
	public void testCircuitBreakerBehaviour() throws InterruptedException {
		CircuitBreakerRegistry registry = CircuitBreakerRegistry.ofDefaults();
		int times = 3;
		int waitSeconds = 2;
		CircuitBreakerConfig config = CircuitBreakerConfig.custom().ringBufferSizeInHalfOpenState(1)
				.ringBufferSizeInClosedState(times).waitDurationInOpenState(Duration.ofMillis(WAIT_TIME)).build();
		CircuitBreaker circuitBreaker = registry.circuitBreaker("something", config);
		for (int i = 0; i < times; i++) {
			assertTrue("Circuit-breaker call should be permitted", circuitBreaker.isCallPermitted());
			circuitBreaker.recordFailure(new RuntimeException("Fail! " + i));
		}
		assertFalse("Circuit-breaker call should not be permitted", circuitBreaker.isCallPermitted());
		Thread.sleep(TimeUnit.SECONDS.toMillis(waitSeconds));
		assertTrue("Circuit-breaker call should be permitted", circuitBreaker.isCallPermitted());
	}
```

whereas this (following your instructions above) doesn't:
```
	@Test
	public void testCircuitBreakerBehaviour() throws InterruptedException {
		CircuitBreakerRegistry registry = CircuitBreakerRegistry.ofDefaults();
		int times = 3;
		int waitSeconds = 2;
		CircuitBreakerConfig config = CircuitBreakerConfig.custom().ringBufferSizeInHalfOpenState(times)
				.failureRateThreshold(100).ringBufferSizeInClosedState(times)
				.waitDurationInOpenState(Duration.ofMillis(WAIT_TIME)).build();
		CircuitBreaker circuitBreaker = registry.circuitBreaker("something", config);
		for (int i = 0; i < times; i++) {
			assertTrue("Circuit-breaker call should be permitted", circuitBreaker.isCallPermitted());
			circuitBreaker.recordFailure(new RuntimeException("Fail! " + i));
		}
		assertFalse("Circuit-breaker call should not be permitted", circuitBreaker.isCallPermitted());
		Thread.sleep(TimeUnit.SECONDS.toMillis(waitSeconds));
		assertTrue("Circuit-breaker call should be permitted", circuitBreaker.isCallPermitted());
	}
```

**Comment by RobWin** [comment]
Thx. You are right. Right now the CB changes to OPEN state when the failure rate is above (>) the threshold, but it should be above or equal (>=).

**Comment by flozano** [comment]
ah! so with 99 it will work as I expect :+1:

**Comment by RobWin** [comment]
But I will change that behavior in the next version.
Your behavior is more intuitive.

**Comment by RobWin** [comment]
I published a `0.5.0-SNAPSHOT` if you want to test the version.

**Comment by RobWin** [comment]
0.5.0 has been released
